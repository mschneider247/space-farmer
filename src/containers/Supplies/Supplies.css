h4 {
  margin-left: 350px;
}

.supply_card_image {
  width: 100%;
  height: 52%;
  border-radius: 20px;
  border: 4px solid darkslategray;
}

.chosen {
  text-align: center;
  margin: 1%;
  padding: 0 2% 2% 2%;
  border: 1px solid black;
  border-radius: 10px;
  width: 200px;
  height: 250px;
  box-shadow: -4px 0px 10px 2px black;
  background: darkslategray;
  color: black;
}

.supplies {
  margin-left: 5%;
  /* margin-bottom: 15%; */
  padding: 5%;
  display: flex;
  flex-wrap: wrap;
}

.supply-card {
  text-align: center;
  margin: 1%;
  padding: 0 2% 2% 2%;
  border: 1px solid darkgreen;
  border-radius: 10px;
  width: 200px;
  height: 250px;
  background: rgba(0, 0, 0, 50%);
}

.supply-card:hover {
  transform: scale(0.99);
  cursor: pointer;
}

.supply-footer {
  width: 300px;
  padding: 1%;
  position: fixed;
  bottom: 5%;
  left: -1%;
  margin-left: 8%;
  display: flex;
  flex-direction: column;
  background-color: #5e5e5e;
  border: 1px solid green;
  border-radius: 10px;
  box-shadow: -4px 0px 10px 2px black;
}

#totalcost, #totalweight{
  color: black;
  font-size: 18px;
  text-align: center;
  background: transparent;
}

#pick_supplies {
  margin-left: 8%;
}

.footer-text, .footer-btn, a {
  background: transparent;
}

.symbols {
  color: lime;
}

.supply-btn {
  width: 100%;
}