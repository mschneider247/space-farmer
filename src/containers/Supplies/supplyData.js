import tonka from '../../images/tonka.jpg';
import mule from '../../images/mule.jpg';
import combine from '../../images/combine.jpg';
import hoe from '../../images/hoe.jpg';
import gardentools from '../../images/gardentools.jpg';
import hal9000 from '../../images/hal9000.jpg';
import dirt from '../../images/dirt.jpg';
import moredirt from '../../images/moredirt.jpg';
import mostdirt from '../../images/mostdirt.jpg';
import carrots from '../../images/carrots.jpg';
import variety from '../../images/variety.jpg';
import exotic from '../../images/exotic.jpg';

const supplyData = [
  {
    id: 1,
    name: 'Tonka Truck',
    cost: 20,
    weight: 2,
    imageURL: tonka,
    isChosen: false,
  },
  {
    id: 2,
    name: 'Mule',
    cost: 150,
    weight: 500,
    imageURL: mule,
    isChosen: false,
  },
  {
    id: 3,
    name: 'Combine',
    cost: 420000,
    weight: 36000,
    imageURL: combine,
    isChosen: false,
  },
  {
    id: 4,
    name: 'One cubic foot of dirt',
    cost: 7,
    weight: 40,
    imageURL: dirt,
    isChosen: false,
  },
  {
    id: 5,
    name: 'One cubic yard of dirt',
    cost: 40,
    weight: 1080,
    imageURL: moredirt,
    isChosen: false,
  },
  {
    id: 6,
    name: '100 cubic yards of dirt',
    cost: 3000,
    weight: 108000,
    imageURL: mostdirt,
    isChosen: false,
  },
  {
    id: 7,
    name: 'Basic hoe',
    cost: 6,
    weight: 3,
    imageURL: hoe,
    isChosen: false,
  },
  {
    id: 8,
    name: 'Garden tools',
    cost: 35,
    weight: 20,
    imageURL: gardentools,
    isChosen: false,
  },
  {
    id: 9,
    name: 'HAL 9000',
    cost: 34000,
    weight: 200,
    imageURL: hal9000,
    isChosen: false,
  },
  {
    id: 10,
    name: 'Carrots',
    cost: 1,
    weight: 1,
    imageURL: carrots,
    isChosen: false,
  },
  {
    id: 11,
    name: 'Fruit and veggie variety',
    cost: 10,
    weight: 4,
    imageURL: variety,
    isChosen: false,
  },
  {
    id: 12,
    name: 'Exotics',
    cost: 110,
    weight: 9,
    imageURL: exotic,
    isChosen: false,
  },
];

export default supplyData;
