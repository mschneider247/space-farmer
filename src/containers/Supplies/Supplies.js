import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import './Supplies.css';
import PropTypes from 'prop-types';
import { addSupplies } from '../../actions';
import supplyData from './supplyData';

export class Supplies extends Component {
  constructor() {
    super();
    this.state = {
      supplies: supplyData,
      supplyCost: 0,
      supplyWeight: 0,
    };
  }

  handleSubmit = () => {
    this.props.addSupplies({ supplyCost: this.state.supplyCost, supplyWeight: this.state.supplyWeight });
    const clearSupplies = this.state.supplies.map(supply => supply.isChosen = false);
    this.setState({ supplies: clearSupplies });
  }

  addOrRemoveSupplyCostandWeight = (supply) => {
    let cost, weight;
    if (supply.isChosen) {
      cost = this.state.supplyCost + supply.cost;
      weight = this.state.supplyWeight + supply.weight;
    } else {
      cost = this.state.supplyCost - supply.cost;
      weight = this.state.supplyWeight - supply.weight;
    }
    this.setState({ supplyCost: cost });
    this.setState({ supplyWeight: weight });
  };

  handleSupplies = (id) => {
    const mappedSupplies = this.state.supplies.map(supply => {
      if (supply.id === id) {
        supply.isChosen = !supply.isChosen;
        this.addOrRemoveSupplyCostandWeight(supply);
      }
      return supply;
    });
    this.setState({ supplies : mappedSupplies });
  }

  render() {
    const supplies = this.state.supplies.map((supply) => {
      let chosen = 'supply-card';
      if (supply.isChosen) {
        chosen += ' chosen';
      }

      return (
        <div className={chosen} key={supply.id} onClick={() => this.handleSupplies(supply.id)}>
          <p>{supply.name}</p>
          <p>$ {supply.cost}</p>
          <p>{supply.weight} lbs</p>
          <img src={supply.imageURL} alt={supply.name}/>
        </div>
      )
    })

    return (
      <section>
        <section>
          <h4>Pick your supplies!</h4>
        </section>
        <section className="supplies">
          {supplies}
        </section>
        <div className="supply-footer">
          <div className="footer-text">
            <p id="totalcost">Total Cost: <span className="symbols">$</span> {this.state.supplyCost}</p>
            <p id="totalweight">
              Total Weight: {this.state.supplyWeight} 
              <span className="symbols"> lbs</span>
            </p>
          </div>
          <div className="footer-btn">
            <Link to='/rockets'><button onClick={this.handleSubmit} className="supply-btn">Continue to Rockets!</button></Link>
          </div>
        </div>
      </section>
    )
  }
};

export const mapDispatchToProps = dispatch => ({
  addSupplies: (supplies) => dispatch(addSupplies(supplies))
})

export default connect(null, mapDispatchToProps)(Supplies);

Supplies.propTypes = {
  addSupplies: PropTypes.func,
  tractorCost: PropTypes.number,
  tractorWeight: PropTypes.number,
  dirtCost: PropTypes.number,
  dirtWeight: PropTypes.number,
  hoeCost: PropTypes.number,
  hoeWeight: PropTypes.number,
  seedCost: PropTypes.number,
  seedWeight: PropTypes.number,
}