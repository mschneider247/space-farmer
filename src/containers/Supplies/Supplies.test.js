import React from 'react';
import { shallow } from 'enzyme';
import { Supplies } from './Supplies';

describe('Supplies container', () => {
  it('Supplies should match its snapshot', () => {
    const wrapper = shallow(<Supplies/>)
    expect(wrapper).toMatchSnapshot();
  });
})