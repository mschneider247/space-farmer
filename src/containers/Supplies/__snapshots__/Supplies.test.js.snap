// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Supplies container Supplies should match its snapshot 1`] = `ShallowWrapper {}`;
